### PR TITLE
Document UNION queries

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: source/devapi/database/dbiterator.rst [hunk 1]
```diff
@@ -206,7 +206,50 @@ It is also possible to add an extra criterion for any `JOIN` clause. You have to
    //            )
 
 
+UNION queries
+^^^^^^^^^^^^^
+
+An union query is an object, wich contains an array of :ref:`sub_queries`. You just have to give a list of Subqueries
```
Comment (cedric-anne) [inline source/devapi/database/dbiterator.rst]:
```suggestion
An union query is an object, which contains an array of :ref:`sub_queries`. You just have to give a list of Subqueries
```

File: source/devapi/database/dbiterator.rst [hunk 2]
```diff
@@ -206,7 +206,50 @@ It is also possible to add an extra criterion for any `JOIN` clause. You have to
...
+you have already prepare, or arrays of parameters that will be used to build them.
```
Comment (cedric-anne) [inline source/devapi/database/dbiterator.rst]:
```suggestion
you have already prepared, or arrays of parameters that will be used to build them.
```

File: source/devapi/database/dbiterator.rst [hunk 3]
```diff
@@ -206,7 +206,50 @@ It is also possible to add an extra criterion for any `JOIN` clause. You have to
...
+   <?php
+   $sub1 = new \QuerySubQuery([
+      'SELECT' => 'field1 AS myfield',
+      'FROM'   => 'table1'
+   ]);
+   $sub2 = new \QuerySubQuery([
+      'SELECT' => 'field2 AS myfield',
+      'FROM'   => 'table2'
+   ]);
+   $union = \QueryUnion([$sub1, $sub2]);
```
Comment (cedric-anne) [inline source/devapi/database/dbiterator.rst]:
```suggestion
   $union = new \QueryUnion([$sub1, $sub2]);
```

File: source/devapi/database/dbiterator.rst [hunk 4]
```diff
@@ -206,7 +206,50 @@ It is also possible to add an extra criterion for any `JOIN` clause. You have to
...
+
+As you can see on the above example, a ``UNION ALL`` query is built. If you want your results to be deduplicated,
+(standard ``UNION``):
+
+.. code-block:: php
+
+  <?php
+   //...
+   //passing true as second argument will activate deduplication.
+   $union = \QueryUnion([$sub1, $sub2], true);
```
Comment (cedric-anne) [inline source/devapi/database/dbiterator.rst]:
```suggestion
   $union = new \QueryUnion([$sub1, $sub2], true);
```

**Comment by trasher** [comment]
Thanks @ddurieux and @cedric-anne :)

**Pull request merged**